src/Sarif.Viewer.VisualStudio/SarifToolWindow.cs

```diff
@@ -46,9 +54,71 @@ public SarifToolWindow() : base(null)
             // the object returned by the Content property.
             Content = new SarifToolWindowControl();
-            Control.DataContext = null;
+            Control.Loaded += this.Control_Loaded;
+            Control.Unloaded += this.Control_Unloaded;
 
-            ResetSelection();
+            IComponentModel componentModel = (IComponentModel)AsyncPackage.GetGlobalService(typeof(SComponentModel));
+            if (componentModel != null)
+            {
+                this.sarifErrorListEventSelectionService = componentModel.GetService<ISarifErrorListEventSelectionService>();
+                this.textViewCaretListenerService = componentModel.GetService<ITextViewCaretListenerService<ITextMarkerTag>>();
+            }
+        }
+
+        private void Control_Loaded(object sender, System.Windows.RoutedEventArgs e)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+            
+            // Subscribe to the error list event service so we
+            // can properly update the data context of this control when the selection changes.
+            if (this.sarifErrorListEventSelectionService != null)
+            {
+                this.sarifErrorListEventSelectionService.NavigatedItemChanged += this.SarifListErrorItemNavigated;
+                this.UpdateDataContext(this.sarifErrorListEventSelectionService.NavigatedItem);
+            }
+
+            // Subscribe to the caret listener service so we can ensure the proper call tree node
+            // is selected when the caret enters a tag representing a call tree node.
+            if (this.textViewCaretListenerService != null)
+            {
+                this.textViewCaretListenerService.CaretEnteredTag += this.TextViewCaretListenerService_CaretEnteredTag;
+            }
+        }
+
+        private void Control_Unloaded(object sender, System.Windows.RoutedEventArgs e)
+        {
+            if (this.sarifErrorListEventSelectionService != null)
+            {
+                this.sarifErrorListEventSelectionService.NavigatedItemChanged -= this.SarifListErrorItemNavigated;
+            }
+
+            if (this.textViewCaretListenerService != null)
+            {
+                this.textViewCaretListenerService.CaretEnteredTag -= this.TextViewCaretListenerService_CaretEnteredTag;
+            }
+        }
+
+        private void TextViewCaretListenerService_CaretEnteredTag(object sender, CaretEventArgs e)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            if (e.Tag.Context is CallTreeNode node)
+            {
+                // If the node is visible in the explorer pane's UI (controlled by verbosity slider)
+                // and have a parent call tree node, then mark it as selected.
+                if (node.Visibility == Visibility.Visible && node.CallTree != null)
+                {
+                    node.CallTree.SelectedItem = node;
```

Comment:
> node.CallTree.SelectedItem = node; [](start = 19, length = 35)

Comment the fact that setting this property causes all of the WPF\XAML in the UI to update.